### PR TITLE
Add parsing numa vmstat

### PR DESCRIPTION
**Pull request by kwisniewski98**
Signed-off-by: Wisniewski, Krzysztof2 <Krzysztof2.Wisniewski@intel.com>

### CODE REVIEW

File: sysfs/vmstat_numa.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/prometheus/procfs/internal/util"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
```
Comment (Creatone) [inline sysfs/vmstat_numa.go:31]: Space after these imports.

File: sysfs/vmstat_numa.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+var nodeNumberRegexp      = regexp.MustCompile(`.*devices/system/node/node([0-9]*)`)
```
Comment (Creatone) [inline sysfs/vmstat_numa.go]: Can you put these vars in a group?

```
var (
	nodeNumberRegexp = regexp.MustCompile(`.*devices/system/node/node([0-9]*)`)
	nodePattern      = "devices/system/node/node[0-9]*"
)
```

File: sysfs/vmstat_numa.go [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sysfs
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/prometheus/procfs/internal/util"
```
Comment (Creatone) [inline sysfs/vmstat_numa.go]: This import should be separated.

File: sysfs/vmstat_numa.go [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+var nodeNumberRegexp      = regexp.MustCompile(`.*devices/system/node/node([0-9]*)`)
+var nodePattern = "devices/system/node/node[0-9]*"
```
Comment (Creatone) [inline sysfs/vmstat_numa.go]: Can you use this pattern in the `nodeNumberRegexp`?

File: sysfs/vmstat_numa.go [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+	NrVmscanWrite              uint64
+	NrVmscanImmediateReclaim   uint64
+	NrDirtied                  uint64
+	NrWritten                  uint64
+	NrKernelMiscReclaimable    uint64
+	NrFollPinAcquired          uint64
+	NrFollPinReleased          uint64
+}
+
+func (fs FS) VmStatNuma() (map[int]VMStat, error ){
```
Comment (Creatone) [inline sysfs/vmstat_numa.go]: Consider using `VMStatNUMA`.

File: sysfs/vmstat_numa.go [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+			return  nil, err
+		}
+		m[nodeNumber] = nodeStats
+
+
+	}
+	return m, nil
+}
+
+func parseVMStatNuma(r []byte ) (VMStat, error) {
```
Comment (Creatone) [inline sysfs/vmstat_numa.go]: Consider using `parseVMStatNUMA`.

File: sysfs/vmstat_numa.go [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+			vmStat.NrDirtied = fv
+		case "nr_written":
+			vmStat.NrWritten = fv
+		case "nr_kernel_misc_reclaimable":
+			vmStat.NrKernelMiscReclaimable = fv
+		case "nr_foll_pin_acquired":
+			vmStat.NrFollPinAcquired = fv
+		case "nr_foll_pin_released":
+			vmStat.NrFollPinReleased = fv
+
```
Comment (Creatone) [inline sysfs/vmstat_numa.go]: Delete this space.

File: sysfs/vmstat_numa_test.go [hunk 8]
```diff
@@ -0,0 +1,31 @@
+package sysfs
+
+import (
+	"testing"
+)
+
+func TestParseVMStatNuma(t *testing.T) {
```
Comment (Creatone) [inline sysfs/vmstat_numa_test.go]: Consider using `TestParseVMStatNUMA`.

File: sysfs/vmstat_numa.go [hunk 9]
```diff
@@ -0,0 +1,240 @@
+// Copyright 2019 The Prometheus Authors
```
Comment (Creatone) [inline sysfs/vmstat_numa.go]: Change year to 2020.

File: sysfs/vmstat_numa.go [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+		file, err := util.ReadFileNoStat(filepath.Join(node, "vmstat"))
+		if err != nil {
+			return  nil, err
+		}
+		nodeStats, err := parseVMStatNuma(file)
+		if err != nil {
+			return  nil, err
+		}
+		m[nodeNumber] = nodeStats
+
```
Comment (Creatone) [inline sysfs/vmstat_numa.go]: Delete these spaces.

File: sysfs/vmstat_numa.go [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+	NrFollPinReleased          uint64
+}
+
+func (fs FS) VmStatNuma() (map[int]VMStat, error ){
+	m := make(map[int]VMStat)
+	nodes, err := filepath.Glob(fs.sys.Path(nodePattern))
+	if err != nil {
+		return nil, err
+	}
+	for _, node := range nodes {
```
Comment (Creatone) [inline sysfs/vmstat_numa.go:95]: Add space above.

File: sysfs/vmstat_numa.go [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+		nodeNumber, err := strconv.Atoi(nodeNumberRegexp.FindStringSubmatch(node)[1])
```
Comment (Creatone) [inline sysfs/vmstat_numa.go]: Unfortunatelly you're not 100% sure that `FindStringSubmatch` would return 2 strings. Separate this and check the length of the returned value.

**Review by Creatone — COMMENTED** [review]
@kwisniewski98 PTAL

File: sysfs/vmstat_numa.go (hunk 4 above)
Comment (kwisniewski98) [reply sysfs/vmstat_numa.go]: This won't work. `filepath.Glob` wont return any results if there is group

File: sysfs/vmstat_numa_test.go [hunk 13]
```diff
@@ -0,0 +1,31 @@
+package sysfs
```
Comment (Creatone) [inline sysfs/vmstat_numa_test.go:14]: Add the license here.

**Review by Creatone — COMMENTED** [review]
Only one nit.

**Pull request closed without merging**